uniflow/flow/config.py

```diff
@@ -133,7 +133,73 @@ class TransformCopyConfig(TransformConfig):
+                    context="Our new business wins are supported by our product leadership strategy of bringing new product to market that provides value for our customers, such as market-leading 500 bar GDi technology, helping customers improve efficiency, reduce emissions and lower costs leveraging our GDi technology and capital to provide a value-focused solution for our off-highway diesel applications and hydrogen ICE that differentiates us from our competition. We're helping our customers move towards carbon neutral and carbon-free fuels with solutions using ethanol, biofuels and hydrogen, as it's our view that a liquefied or gaseous fuel is going to be a key element of our journey to carbon neutrality.",
+                    answer=["500 bar GDi technology", "carbon neutral"]
+                ),
+                    Context(
+                    context="The Eiffel Tower, located in Paris, France, is one of the most famous landmarks in the world. It was constructed in 1889 and stands at a height of 324 meters.",
+                    answer=[],
+                ),
+            ],
+        )
+    )
+
+@dataclass
+class TransformForClusteringOpenAIGPT4Config:
+    flow_name: str = "TransformOpenAIFlow"
+    model_config: ModelConfig = field(
+        default_factory=lambda: OpenAIModelConfig(
+            model_name="gpt-4",
+            model_server="OpenAIModelServer",
+            num_call=1,
+            temperature=0,
+            response_format={"type": "json_object"}
+        )
+    )
+    prompt_template: PromptTemplate = field(
+        default_factory=lambda: PromptTemplate(
+            instruction="""
+                As an expert in cutting-edge technologies, your mission is to analyze a list of technology-related terms. Your primary goal is twofold: firstly, to cluster these terms into distinct groups based on their semantic similarities, where each group encapsulates a specific technological concept. Secondly, within these clusters, identify and merge terms that are essentially synonymous. Your final output should be a well-structured dictionary. In this dictionary, each key signifies a unique category of technology, and its corresponding value is a list of technology terms.
+            """,
 
+            few_shot_prompt = [
+                Context(
+                    context=["artificial intelligence", "AI", "500 bar GDi technology", "ML", "500 bar GDi", "machine learning"],
+                    answer={
+                        "500_BAR_GDI": [ "500 bar GDi"],
+                        "AIML": ["AI", "ML"],
+                    }
+                ),
+                Context(
+                    context=["cryptocurrency", "blockchain", "Bitcoin", "Ethereum", "digital currency", "crypto mining"],
+                    answer={
+                        "CRYPTO_CURRENCY": ["cryptocurrency", "Bitcoin", "Ethereum", "digital currency"],
+                        "BLOCKCHAIN_TECH": ["blockchain", "crypto mining"],
```

Comment:
this example is not well defined, since Ethereum is the blockchain and Ether/ETH is the currency. Can you add some other orthogonal techs e.g. "mRNA vaccine"?

Reply:
Modified, Thanks